app/src/main/rust/src/img/utils.rs

```diff
@@ -0,0 +1,5 @@
+use image::{Luma, Pixel, Primitive};
+
+pub fn to_luma8<S: Primitive, P: Pixel<Subpixel = S>>(p: &P) -> Luma<u8> {
+    Luma([p.to_luma().0[0].to_u8().unwrap()])
```

Comment:
Is this even correct for non-RGBA8888?

Reply:
> then cast to LumaU8 using num-traits

That doesn't seem right. Won't truncation lead to an inaccurate value?

Reply:
https://github.com/image-rs/image/issues/1212
We might need to copy the conversion code.